### ISSUE TITLE
Get CrossEntropy working

### ISSUE THREAD

**Issue by bclarkson-code** [bclarkson-code/Tricycle#2]
The crossentropy loss function does not currently work as expected and its test fails. The reason should be investigated and fixed

**Comment by bclarkson-code** [comment]
After several rewrites, it was discovered that vectorising was being done too soon
